src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXCEPTION = "Wrong index - ";
+    private Node first;
+    private Node last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node newNode = new Node(value);
+        if (size == 0) {
+            first = newNode;
+            last = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(EXCEPTION + index);
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node newNode = new Node(value);
+            Node current = getNode(index);
+
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev = newNode;
+
+            if (newNode.prev != null) {
+                newNode.prev.next = newNode;
+            } else {
+                first = newNode;
+            }
+            size++;
```

Comment:
`size++` only for else case?

Reply:
yes, it is. because when adding an item to the end or when indexing the list, you need to update the size value